src/context/WalletProvider/KeepKey/components/Label.tsx

```diff
@@ -1,19 +1,60 @@
 import { Button, Input, ModalBody, ModalHeader } from '@chakra-ui/react'
+import { useToast } from '@chakra-ui/toast'
+import { RecoverDevice, ResetDevice } from '@shapeshiftoss/hdwallet-core'
 import { useRef, useState } from 'react'
-import { useHistory } from 'react-router-dom'
+import { useTranslate } from 'react-polyglot'
 import { Text } from 'components/Text'
-import { KeepKeyRoutes } from 'context/WalletProvider/routes'
+import { useWallet } from 'hooks/useWallet/useWallet'
 
 export const KeepKeyLabel = () => {
   const [loading, setLoading] = useState(false)
-  const history = useHistory()
-
+  const {
+    setDeviceState,
+    state: {
+      deviceState: { disposition, recoverWithPassphrase, recoveryEntropy },
+      wallet,
+    },
+  } = useWallet()
+  const toast = useToast()
+  const translate = useTranslate()
   const inputRef = useRef<HTMLInputElement | null>(null)
 
-  const handleSubmit = async () => {
+  const handleInitializeSubmit = async () => {
+    setLoading(true)
+    const label = inputRef.current?.value
+    const resetMessage: ResetDevice = { label: label ?? '', pin: true }
+    setDeviceState({ awaitingDeviceInteraction: true })
+    await wallet?.reset(resetMessage).catch(e => {
+      console.error(e)
+      toast({
+        title: translate('common.error'),
+        description: e?.message ?? translate('common.somethingWentWrong'),
+        status: 'error',
+        isClosable: true,
+      })
+    })
+  }
+
+  const handleRecoverSubmit = async () => {
     setLoading(true)
     const label = inputRef.current?.value
-    history.push({ pathname: KeepKeyRoutes.NewRecoverySentence, state: { label } })
+    setDeviceState({ awaitingDeviceInteraction: true })
+    const recoverParams: RecoverDevice = {
+      entropy: parseInt(recoveryEntropy) as RecoverDevice['entropy'],
```

Comment:
we could write a `parseIntToEntropy` with a type predicate to avoid casting here, otherwise approved

Reply:
That cast did feel gross. Fixed in https://github.com/shapeshift/web/pull/1561/commits/b4a1ea2ad9eacd8b2f1d1fc8f174add81e3b1fe0.